bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/handler/PentairIntelliChlorHandler.java

```diff
@@ -34,91 +37,154 @@
                 break;
-        }
 
-        logger.debug("Intellichlor command: {}", pic);
-    }
-
-    /**
-     * Helper function to compare and update channel if needed. The class variables p29_cur and phsp_cur are used to
-     * determine the appropriate state of the channel.
-     *
-     * @param channel name of channel to be updated, corresponds to channel name in {@link PentairBindingConstants}
-     * @param p Packet representing the former state. If null, no compare is done and state is updated.
-     */
-    public void updateChannel(String channel, PentairPacket p) {
-        PentairPacketIntellichlor pic = (PentairPacketIntellichlor) p;
-
-        switch (channel) {
-            case INTELLICHLOR_SALINITY:
-                if (pic == null || (pic.salinity != pic4cur.salinity)) {
-                    updateState(channel, new DecimalType(pic4cur.salinity));
-                }
-                break;
-            case INTELLICHLOR_SALTOUTPUT:
-                if (pic == null || (pic.saltoutput != pic3cur.saltoutput)) {
-                    updateState(channel, new DecimalType(pic3cur.saltoutput));
-                }
+            case 0x11: // set salt output % command
+                saltOutput = pic.getSaltOutput();
+                updateChannel(INTELLICHLOR_SALTOUTPUT, saltOutput);
+                logger.debug("Intellichlor set output % {}", saltOutput);
                 break;
+            case 0x12: // response to set salt output
+                salinity = pic.getSalinity();
+
+                ok = pic.getOk();
+                lowFlow = pic.getLowFlow();
+                lowSalt = pic.getLowSalt();
+                veryLowSalt = pic.getVeryLowSalt();
+                highCurrent = pic.getHighCurrent();
+                cleanCell = pic.getCleanCell();
+                lowVoltage = pic.getLowVoltage();
+                lowWaterTemp = pic.getLowWaterTemp();
+
+                updateChannel(INTELLICHLOR_SALINITY, salinity);
+                updateChannel(INTELLICHLOR_OK, ok);
+                updateChannel(INTELLICHLOR_LOWFLOW, lowFlow);
+                updateChannel(INTELLICHLOR_LOWSALT, lowSalt);
+                updateChannel(INTELLICHLOR_VERYLOWSALT, veryLowSalt);
+                updateChannel(INTELLICHLOR_HIGHCURRENT, highCurrent);
+                updateChannel(INTELLICHLOR_CLEANCELL, cleanCell);
+                updateChannel(INTELLICHLOR_LOWVOLTAGE, lowVoltage);
+                updateChannel(INTELLICHLOR_LOWWATERTEMP, lowWaterTemp);
+
+                String status = String.format(
+                        "saltoutput = %d, salinity = %d, ok = %b, lowflow = %b, lowsalt = %b, verylowsalt = %b, highcurrent = %b, cleancell = %b, lowvoltage = %b, lowwatertemp = %b",
+                        saltOutput, salinity, ok, lowFlow, lowSalt, veryLowSalt, highCurrent, cleanCell, lowVoltage,
+                        lowWaterTemp);
+                logger.debug("IntelliChlor salinity/status: {}, {}", salinity, status);
```

Comment:
String formatting is an expensive operation so avoid performing it if you don't need to.

```suggestion
                if(logger.isDebugEnable()){
                    String status = String.format(
                        "saltoutput = %d, salinity = %d, ok = %b, lowflow = %b, lowsalt = %b, verylowsalt = %b, highcurrent = %b, cleancell = %b, lowvoltage = %b, lowwatertemp = %b",
                        saltOutput, salinity, ok, lowFlow, lowSalt, veryLowSalt, highCurrent, cleanCell, lowVoltage,
                        lowWaterTemp);
                    logger.debug("IntelliChlor salinity/status: {}, {}", salinity, status);
                }
```